githubmarkdownui/blocks/leaf.py

```diff
@@ -0,0 +1,28 @@
+from typing import Optional
+
+
+def thematic_break() -> str:
+    """Returns a <hr> tag, used to create a thematic break. Equivalent to --- in Markdown."""
+    pass
+
+
+def code_block(text: str, language: Optional[str] = None) -> str:
+    """Creates a code block using HTML syntax. If language is given, then the code block will be
+    created using Markdown syntax, but it cannot be used inside a table.
```

Comment:
Is there a plan to validate the output some how? Probably good old exception. Say that when a user uses the `code_black` with language which produces the Markdown syntax, but then use it inside a table. If so, can we create an issue to track that?

Reply:
I don't think we _can_ validate something like that. I imagine this function would return the given text wrapped in `<pre>` tags if `language` is not given, and use ` ``` ` if `language` is given. 

The user is free to put whatever string they want into the table, and a naive way to check that a code block is being used is to check for the string ` ``` ` in the text the user passes into the table. But this would cause a problem if the user actually wanted three backticks to show up in their table.

I think the only way to enforce that Markdown syntax is not being used in a table is to create classes for the structures that can only be represented with Markdown and check if that is being passed into a table, but I think this would ultimately be more confusing as the user would need to remember to pass in a class for certain cases and not a string, which feels strange especially since the rest of this library (except for lists) takes strings as inputs.

Reply:
That makes sense. If user doesn't have to call a function to compile the whole page, I don't think there's a good place to validate the output.